src/components/views/rooms/RoomSettings.js

```diff
@@ -102,6 +147,8 @@ module.exports = React.createClass({
     componentWillUnmount: function() {
         ScalarMessaging.stopListening();
 
+        MatrixClientPeg.get().removeListener("RoomMember.membership", this._onRoomMemberMembership);
```

Comment:
needs a null-guard to handle logout

or better yet, use the matrixclient from the react context rather than MatrixClientPeg